app/src/utils/validate.js

```diff
@@ -1,13 +1,22 @@
+export function validateIpAddress(input) {
+  const ipFormat = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
+  //type coerce match array or null value to bool
+  return !!input.match(ipFormat);
+}
+
 export const QSLRegEx = /([a-zA-Z0-9]+)\[?(?:(@[()",@$=><!a-zA-Z0-9\-.|&:_]*|\**|\$\$[a-zA-Z0-9,=]+))\]?\{([*|[,@"=a-zA-Z0-9-]*)/;
+export function validateQslQuery(input) {
+  //type coerce match array or null value to bool
+  return !!input.match(QSLRegEx);
+}
 
-export function validateIPaddress(ipaddress) {
-  const ipformat = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
-  if (ipaddress.match(ipformat)) {
-    return true;
-  }
-  return false;
+export function validateHexId(input) {
+  const hexIdFormat = /(0x|0X)?[a-fA-F0-9]+$/g;
+  //type coerce match array or null value to bool
+  return !!input.match(hexIdFormat);
 }
 
+//TODO:DM+FZ - determine if the following util fns belong here, not strictly validations
```

Comment:
exactly. I thought the same :) was focus on features only, did not consider too much.

Reply:
yeah, that makes sense... the project org and structure isn't too bad, but we can still improve it when we've got some time to pay down tech debt